docs/sources/operators-guide/configure/setting-helm-ha-deduplication-consul/index.md

```diff
@@ -0,0 +1,138 @@
+
+Grafana Mimir can deduplicate data from high-availability (HA) Prometheus setup. In this guide, you will see how to configure
+the deduplication for Grafana Mimir helm deployment using external Consul. You also can review 
+[Configuring High Availability]({{< relref "../configuring-high-availability-deduplication.md" >}}) document for some 
+conceptual background.
+
+## Before you begin
+
+You need to have a Grafana Mimir installed.
+
+Refer to [Configuring High Availability]({{< relref "../configuring-high-availability-deduplication.md" >}}) documents
+for high level description on the concept how Mimir deduplicate incoming HA samples.
+
+You also have to configure HA for Prometheus or Grafana Agent. Last, you need a KV store. In this guide you will
+use Consul. You will be guided on the setup if you haven't had one.
+
+## Configure Prometheus or Grafana Agent to send HA external labels
+
+Configure Prometheus or Grafana Agent HA setup by setting label called `cluster` and `__replica__`. These two labels
+name are default labels for HA setup in Grafana Mimir. If you want to change the labels, make sure to change it in Mimir
+too to make Grafana Mimir and Prometheus/Grafana Agent config matches with each other, otherwise HA deduplication
+wouldn't work. `cluster` label value must be same across replica belong to the same cluster. `__replica__` must be
+unique across different replica in the cluster.
+
+```yaml
+global:
+  scrape_interval: 1m
+  scrape_timeout: 10s
+  evaluation_interval: 1m
+  external_labels:
+    __replica__: replica-1
+    cluster: my-prometheus
+```
+
+Reload or restart Prometheus or Grafana Agent after updating the configuration.
+
+## Install Consul using Helm
+
+Install Consul in Kubernetes using
+[Consul helm chart](https://github.com/hashicorp/consul-k8s/tree/main/charts/consul). Follow the documentation to get
+the chart and installing a Consul release. Put a note on the Consul endpoint, because you will need it for Mimir
+configuration.
+
+## Setup Mimir HA config
+
+You can set Mimir HA deduplication configuration in global level or tenant level.
+
+### HA Deduplication Global
+
+You need a Mimir setup installed by Helm. Add the following configuration to your `custom.yaml` to configure HA
+deduplication globally.
+
+```yaml
+# other configurations above
+mimir:
+  structuredConfig:
+    limits:
+      accept_ha_samples: true
+      # The following two configs must match with external_labels config in Prometheus
+      # The config values below are the default and can be removed if you don't want to override to a new value
+      ha_cluster_label: cluster
+      ha_replica_label: __replica__
+    distributor:
+      ha_tracker:
+        enable_ha_tracker: true
+        kvstore:
+          store: consul
+          consul:
+            host: <consul-endpoint> # example: http://consul.consul.svc.cluster.local:8500
+```
+
+Upgrade the Mimir's helm release using the above configuration.
```

Comment:
```suggestion
Upgrade the Mimir's helm release using the preceding configuration.
```

Reply:
The "following" code block is not a configuration but is a command. Hence I changed this to:

```
2. Upgrade the Mimir's helm release using the following command:
```